### PR TITLE
feat: update styling of property pane

### PR DESCRIPTION
**Pull request by tilmx**
Making the property pane and all the property types really cool.

## Tasks
- [x] Styling of String Property
- [x] Styling of Boolean Property
- [x] Styling of Enum Property

## Upcoming PR
- [ ] Styling of Asset Property
- [ ] Styling of Unknown Property

### CODE REVIEW

File: src/lsg/patterns/property-items/boolean-item/index.tsx [hunk 1]
```diff
@@ -15,62 +15,77 @@ interface IndicatorProps {
...
-	box-shadow: inset 0 0 0 ${indicatorBorderWidth}px ${colors.grey60.toString()};
-
-	&:after {
-		content: '';
+	box-shadow: inset 0 0 0 1px ${colors.grey80.toString()};
+	transition: all ease-in-out 0.1s;
+	user-select: none;
+	&::after {
+		position: absolute;
+		content: url('data:image/svg+xml; utf8, <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 12"><path d="M6,6L4.5,7.5L6,6L4.5,4.5L6,6z M6,6l1.5-1.5L6,6l1.5,1.5L6,6z" fill="none" stroke="${colors.grey60.toString()}" stroke-width="1" stroke-linecap="round" stroke-linejoin="round" /></svg>');
```
Comment (marionebl) [inline src/lsg/patterns/property-items/boolean-item/index.tsx]: We might want to move this to `lsg/patterns/icons` and create the inline string with something like this (pseudo code):

```js
const unchecked = ReactDOM.renderToString(<Icon name="x"/>);
const checked = ReactDOM.renderToString(<Icon name="ok"/>);

content: url('data:image/svg+xml,${unchecked}')
{ /* ... */ }
content: url('data:image/svg+xml,${checked}');
```

File: src/lsg/patterns/property-items/boolean-item/index.tsx [hunk 2]
```diff
@@ -15,62 +15,77 @@ interface IndicatorProps {
...
-			background: ${colors.blue40.toString()};
-			box-shadow: inset 0 0 0 ${indicatorBorderWidth}px ${colors.blue40.toString()};
-
-			&:after {
-				transform: translateX(${indicatorWidth - indicatorHeight}px);
+			background: ${colors.blue80.toString()};
+			box-shadow: inset 0 0 0 0.5px ${colors.blue40.toString()};
+			&::after {
+				content: url('data:image/svg+xml; utf8, <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 12"><polyline points="3.8,6 5.2,7.5 8.2,4.5 " fill="none" stroke="${colors.blue40.toString()}" stroke-width="1" stroke-linecap="round" stroke-linejoin="round"></polyline></svg>');
+				transform: translateX(0px);
```
Comment (marionebl) [inline src/lsg/patterns/property-items/boolean-item/index.tsx]: This is a noop? Does it change anything if this is removed?

File: src/lsg/patterns/property-items/boolean-item/index.tsx [hunk 3]
```diff
@@ -15,62 +15,77 @@ interface IndicatorProps {
...
 			? `
-			background: ${colors.blue40.toString()};
-			box-shadow: inset 0 0 0 ${indicatorBorderWidth}px ${colors.blue40.toString()};
-
-			&:after {
-				transform: translateX(${indicatorWidth - indicatorHeight}px);
+			background: ${colors.blue80.toString()};
+			box-shadow: inset 0 0 0 0.5px ${colors.blue40.toString()};
+			&::after {
+				content: url('data:image/svg+xml; utf8, <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 12"><polyline points="3.8,6 5.2,7.5 8.2,4.5 " fill="none" stroke="${colors.blue40.toString()}" stroke-width="1" stroke-linecap="round" stroke-linejoin="round"></polyline></svg>');
```
Comment (marionebl) [inline src/lsg/patterns/property-items/boolean-item/index.tsx]: See comment https://github.com/meetalva/alva/pull/436/files#diff-ab621269c66bbdd4cedbb4074cb5d20dR40

File: src/lsg/patterns/property-items/boolean-item/index.tsx [hunk 4]
```diff
@@ -15,62 +15,77 @@ interface IndicatorProps {
...
 
 const StyledLabel = styled.span`
-	display: block;
+	display: inline-block;
 	font-size: 12px;
 	font-family: ${fonts().NORMAL_FONT};
-	color: ${colors.grey36.toString()};
-	margin-bottom: ${getSpace(SpaceSize.XS)}px;
+	color: ${colors.grey50.toString()};
+	padding: ${getSpace(SpaceSize.XS) + getSpace(SpaceSize.XXS)}px 0 0;
```
Comment (marionebl) [inline src/lsg/patterns/property-items/boolean-item/index.tsx]: Instead if moving the label to center via `padding` we should set `align-items: center` on the the flex container

Comment (marionebl) [reply src/lsg/patterns/property-items/boolean-item/index.tsx]: Also a padding to the left and `box-sizing:  border-box` could avoid the missing visual distance here:

```css
box-sizing: border-box;
padding-right: ${getSpace(SpaceSize.XS)};
```
**before**
screen shot 2018-05-08 at 10 28 46

**after**
screen shot 2018-05-08 at 10 38 43

Comment (marionebl) [reply src/lsg/patterns/property-items/boolean-item/index.tsx]: We'l also want to add a `title` attribute to the spans to make the labels readable via a tooltip:

screen shot 2018-05-08 at 10 41 03

File: src/lsg/patterns/property-items/boolean-item/index.tsx [hunk 5]
```diff
@@ -15,62 +15,77 @@ interface IndicatorProps {
...
+			&:hover {
+				&::after {
+					box-shadow: inset 0 0 0 1px ${colors.blue40.toString()}, 0.5px 0.5px 3px ${colors.blue40.toString()};
+				}
 			}
 		`
 			: ''};
 `;
 
 const StyledLabel = styled.span`
```
Comment (marionebl) [inline src/lsg/patterns/property-items/boolean-item/index.tsx]: Move this to a shared file

File: src/lsg/patterns/property-items/enum-item/index.tsx [hunk 6]
```diff
@@ -18,62 +19,99 @@ export interface EnumItemProps {
...
+
+const StyledSelectWrapper = styled.div`
+	width: 70%;
+	height: 30px;
+	position: relative;
+	display: inline-block;
+	box-sizing: border-box;
+	border-radius: 3px;
+	background-color: ${colors.white.toString()};
+	transition: all 0.1s;
```
Comment (marionebl) [inline src/lsg/patterns/property-items/enum-item/index.tsx]: specify transition properties explicitly

File: src/lsg/patterns/property-items/boolean-item/index.tsx [hunk 7]
```diff
@@ -15,62 +15,77 @@ interface IndicatorProps {
...
 		border-radius: 100%;
 		background: ${colors.white.toString()};
-		transition: all ease-in-out 0.25s;
+		transition: all ease-in-out 0.1s;
 		box-sizing: border-box;
+		box-shadow: inset 0 0 0 1px ${colors.grey60.toString()};
+	}
+	&:hover {
+		&::after {
+			box-shadow: inset 0 0 0 1px ${colors.grey60.toString()}, 0.5px 0.5px 3px ${colors.grey60.toString()};
```
Comment (tilmx) [inline src/lsg/patterns/property-items/boolean-item/index.tsx]: 0.5px/1px media query

File: src/lsg/patterns/property-items/boolean-item/index.tsx (hunk 5 above)
Comment (tilmx) [reply src/lsg/patterns/property-items/boolean-item/index.tsx]: I personally think it’s more effort to understand or change something on a specific property item here if we move the box-shadow line to a separate file (compared to the non-redundancy).

**Comment by tilmx** [comment]
@marionebl addressed all comments

File: src/lsg/patterns/property-items/string-item/index.tsx [hunk 8]
```diff
@@ -60,16 +57,14 @@ export const StringItem: React.StatelessComponent<StringItemProps> = props => {
...
-					type="textarea"
-					value={value}
-					placeholder="Type in"
-				/>
-			</label>
+			<PropertyLabel label={label} />
+			<StyledInput
+				onChange={onChange}
+				onBlur={onBlur}
+				type="textarea"
```
Comment (marionebl) [inline src/lsg/patterns/property-items/string-item/index.tsx]: I think this should be `"text"`. If you want to use a `<textarea/>` here we'll have to create a distinct element for that.

**Comment by tilmx** [comment]
Done!

**Pull request merged**